service-base/src/main/java/org/eclipse/hono/service/plan/PrometheusBasedResourceLimitChecks.java

```diff
@@ -0,0 +1,137 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.hono.service.plan;
+
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.client.HttpResponse;
+import io.vertx.ext.web.client.WebClient;
+import io.vertx.ext.web.client.predicate.ResponsePredicate;
+import org.eclipse.hono.service.metric.MicrometerBasedMetrics;
+import org.eclipse.hono.util.TenantObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * An implementation of {@link ResourceLimitChecks} which uses metrics data from the prometheus backend to check if
+ * further connections or messages are allowed by comparing with the configured limits.
+ */
+@Component
+public final class PrometheusBasedResourceLimitChecks implements ResourceLimitChecks {
+
+    private final Logger log = LoggerFactory.getLogger(getClass());
+    private final WebClient client;
+    private PrometheusBasedResourceLimitChecksProperties config;
+
+    /**
+     * Creates a new PrometheusBasedResourceLimitChecks instance.
+     *
+     * @param vertx The Vertx instance to use.
+     * @throws NullPointerException if any of the parameters are {@code null}.
+     */
+    public PrometheusBasedResourceLimitChecks(final Vertx vertx) {
+        this.client = WebClient.create(Objects.requireNonNull(vertx));
+    }
+
+    @Override
+    public Future<?> isConnectionLimitExceeded(final TenantObject tenant) {
+        Objects.requireNonNull(tenant);
+        log.trace("Connections limit for tenant [{}] is [{}]", tenant.getTenantId(), tenant.getConnectionsLimit());
+        if (tenant.getConnectionsLimit() != -1) {
+            return queryForSumInPrometheus(MicrometerBasedMetrics.METER_CONNECTIONS_AUTHENTICATED.replace(".", "_"),
```

Comment:
the metric name should be configurable and have a reasonable default value which is not computed on every invocation ...

Reply:
@sophokles73 Do you want it to be configurable by the user as Prometheus _host_ and _port_?

Reply:
That would make sense, wouldn't it?

Reply:
@sophokles73 The metrics names in hono are clearly defined as in the [hono-docs](https://www.eclipse.org/hono/api/metrics/). IMHO making the metric name configurable won't be that much useful. If any user decides to implement his/her own metrics (for the number of connections) with a different metric name instead of the standard one shipped with hono, then he/she could use this feature.

Reply:
Can we at least agree that the name should not be re-computed on every invocation?

Reply:
@sophokles73 That for sure 👍